src/helpers/tracks-file-parser/worker/tracks-file-parser-worker.ts

```diff
@@ -36,48 +36,44 @@ const parseTrack = async (
     return null
   }
 
-  const fileBuffer = await new Response(file).arrayBuffer()
-  const fileUint8 = new Uint8Array(fileBuffer)
-  const tags = parseMetadata(fileUint8)
+  try {
+    const fileBuffer = await new Response(file).arrayBuffer()
+    const fileUint8 = new Uint8Array(fileBuffer)
+    const tags = await parseMetadata(
+      fileUint8,
+      { mimeType: file.type, path: file.name, size: file.size },
+      { duration: true },
+    )
+
+    let imageBlob
+    if (tags.common.picture?.length) {
+      const [image] = tags.common.picture
+      const imageData = new Uint8ClampedArray(image.data)
+      imageBlob = new Blob([imageData], { type: image.type })
+    }
 
-  if (!tags) {
+    const artists =
+      getValueOrUndefined(tags.common.artist)
+        ?.split(/,|&/)
```

Comment:
I strongly recommand not to split this. If you want individual artists check tags.common.artists

Reply:
I didn't want to change the current logic of this and I don't know if the artists property does the same(-ish) thing that this code part does. Can you tell me how the artists are generated or link me to the specific code block that does that? I think music metadata formats mostly have just an artist property.

Reply:
I just tested this with few of my local files [here](https://stackblitz.com/edit/js-xyt4ot?file=index.js) and mm doesn't actually return separate artists just same string as first array item, it's very likely that my mp3 files are incorrectly encoded, but I imagine a lot of peoples are then.

Reply:
By splitting it yourself you will split valid artist names into multiple artists which are not multiple artist. [Bob Markey & The Wailers](https://musicbrainz.org/artist/c296e10c-110a-4103-9e77-47bfebb7fb2e) is one artist (a group). Also [AC/DC](https://musicbrainz.org/artist/66c662b6-6e2f-4930-8610-912e24c63ed1) is one group, and so is [love | less](https://musicbrainz.org/artist/f95fe8a5-19d0-4222-be08-c55dba7ffd47).

Reply:
I looked into how other players deal with this, and they don't split artists either, so I am in favor of this. After this change if "Bob Markey" has any solo tracks it will show as separate entry in artists tab from "Bob Markey & The Wailers", which I think is fine.

Reply:
I changed this now. I also think it is better this way, but I didn't want to change the way @minht11 manages/views his music collection.